src/entities/base.js

```diff
@@ -0,0 +1,19 @@
+class Base {
+    constructor(attr) {
+        this.attr = attr
```

Comment:
Do I understand correctly that this will not work if I need to add multiple properties to the class instance?
For example User class and maybe I want to add Name and Last name right from the `new User`.
This will not work, right?
What would I need to do to fix that?
Would I need to give the User class its own `constructor`?

Reply:
You may overwrite constructor in `User` class to call it like this
```javascript
new User({ name: 'Alex Senyuk'}, 'Alex', 'Senyuk')
```
But this solution is bad

Better to have 
```javscript
new User({ firstName: 'Alex', lastName: 'Senyuk' })
```
In this case you don't need to overwrite constuctor, just add 2 get method and maybe change `name` getter
```javascript
get name() {
  return `${this.firstName} ${this.lastName}`
}
```

Reply:
So the current constructor will be ok with `new User({ firstName: 'Alex', lastName: 'Senyuk' })`?

Do you mean I'd need to have individual getters for each property?

Reply:
in this case yes, and for any other virtual properties too. (like here)
``` javascript 
get name() {
  return `${this.firstName} ${this.lastName}`
}
```